app/controllers/categories_controller.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+# CategoriesController
+class CategoriesController < ApplicationController
+  include ApplicationHelper
+  before_action :init_values, only: %i[index show]
+
+  def index
+    render :show
+  end
+
+  def show; end
+
+  private
+
+  def init_values
+    @active_more_btn = false
+    @page_number = 1
+    load_books_with_category(params) if params[:id]
+    load_books_without_category(params) unless params[:id]
+    show_more_button?(params)
+  end
+
+  def load_books_without_category(params)
+    @category = nil
+    init_page_number(params)
+    init_filter(params)
+    @books = Book.by_filter(@filter, @page_number)
+  end
+
+  def load_books_with_category(params)
+    @category = Category.find(params[:id])
+    category_id = params[:id]
+    init_page_number(params)
```

Comment:
Move `init_page_number`, `init_filter`, and `@active_more_btn` assignment to the presenter: https://github.com/andypike/rectify#presenters